src/WebJobs.Script.WebHost/ProxyFunctionExecutor.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web.Http.Dependencies;
+using Microsoft.Azure.AppService.Proxy.Client.Contract;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Azure.WebJobs.Script.WebHost;
+using Microsoft.Azure.WebJobs.Script.WebHost.Controllers;
+using Microsoft.Azure.WebJobs.Script.WebHost.WebHooks;
+
+namespace Microsoft.Azure.WebJobs.Script.Host
+{
+    public class ProxyFunctionExecutor : IFuncExecutor
+    {
+        private readonly WebScriptHostManager _scriptHostManager;
+        private readonly IDependencyResolver _dependencyResolver;
+        private FunctionsController _functionsController;
+
+        internal ProxyFunctionExecutor(WebScriptHostManager scriptHostManager, IDependencyResolver dependencyResolver, FunctionsController functionsController)
+        {
+            _scriptHostManager = scriptHostManager;
+            _dependencyResolver = dependencyResolver;
+            _functionsController = functionsController;
+        }
+
+        public async Task ExecuteFuncAsync(string funcName, Dictionary<string, object> arguments, CancellationToken cancellationToken)
+        {
+            HttpRequestMessage request = arguments[ScriptConstants.AzureFunctionsHttpRequestKey] as HttpRequestMessage;
+            var function = _scriptHostManager.GetHttpFunctionOrNull(request);
+            if (function == null)
+            {
+                // request does not map to an HTTP function
+                request.Properties[ScriptConstants.AzureFunctionsHttpResponseKey] = new HttpResponseMessage(HttpStatusCode.NotFound);
+                return;
+            }
+            request.SetProperty(ScriptConstants.AzureFunctionsHttpFunctionKey, function);
+
+            var authorizationLevel = await FunctionsController.DetermineAuthorizationLevelAsync(request, function, _dependencyResolver);
+            if (function.Metadata.IsExcluded ||
```

Comment:
Can we remove the `IsExcluded` and `IsDisabled` checks here? I think these should already be handled in `TryParseFunctionMetadata` - the functions won't be added to the script host function collection.

Reply:
IsExcluded can be removed. but not the check for IsDisabled because they can still be  called with admin level auth.

---
In reply to: [135930304](https://github.com/Azure/azure-webjobs-sdk-script/pull/1839#discussion_r135930304) [](ancestors = 135930304)